apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/context/RuntimeContext.java

```diff
@@ -20,19 +20,28 @@
 
 import java.util.Map;
 import java.util.concurrent.ConcurrentHashMap;
+import org.apache.skywalking.apm.agent.core.logging.api.ILog;
+import org.apache.skywalking.apm.agent.core.logging.api.LogManager;
 
 /**
  * RuntimeContext is alive during the tracing context.
  * It will not be serialized to the collector, and always stays in the same context only.
  *
  * In most cases, it means it only stays in a single thread for context propagation.
  *
- * @author wusheng
+ * @author wusheng, ascrutae
  */
 public class RuntimeContext {
+    private ILog logger = LogManager.getLogger(RuntimeContext.class);
+    private final ThreadLocal<RuntimeContext> contextThreadLocal;
     private Map context = new ConcurrentHashMap(0);
 
+    public RuntimeContext(ThreadLocal<RuntimeContext> contextThreadLocal) {
+        this.contextThreadLocal = contextThreadLocal;
+    }
+
     public void put(Object key, Object value) {
+        logger.debug("Storage Key[{}] into runtime context.", key);
```

Comment:
Why do you need to debug this? Our release agent config log.level=debug. I can see this would be very dangerous, which affects our performance in default mode.